### PR TITLE
fix: restore python linting.

### PR DESCRIPTION
**Pull request by iFurySt**
Since we used the diff to transition the lint in #1457, I think the transition is almost done. So we should bring the Python lint back on track.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -47,11 +47,4 @@ jobs:
...
       - name: Run pre-commit hooks
-        if: github.ref != 'refs/heads/main'
-        run: |
-          git fetch https://github.com/OpenDevin/OpenDevin.git main:main && \
-          pre-commit run \
-            --files \
-            $(git diff --name-only $(git merge-base main $(git branch --show-current)) $(git branch --show-current) | tr '\n' ' ') \
-            --show-diff-on-failure \
-            --config ./dev_config/python/.pre-commit-config.yaml
+        run: pre-commit run --files opendevin/**/* agenthub/**/* --show-diff-on-failure --config ./dev_config/python/.pre-commit-config.yaml
```
Comment (li-boxuan) [inline .github/workflows/lint.yml:50]:
```suggestion
        run: pre-commit run --files opendevin/**/* agenthub/**/* evaluation/**/* --show-diff-on-failure --config ./dev_config/python/.pre-commit-config.yaml
```

File: Makefile [hunk 2]
```diff
@@ -172,7 +172,7 @@ install-precommit-hooks:
 
 lint-backend:
 	@echo "$(YELLOW)Running linters...$(RESET)"
-	@poetry run pre-commit run --files $$(git diff --name-only $$(git merge-base main $$(git branch --show-current)) $$(git branch --show-current) | tr '\n' ' ') --show-diff-on-failure --config $(PRECOMMIT_CONFIG_PATH)
+	@poetry run pre-commit run --files opendevin/**/* agenthub/**/* --show-diff-on-failure --config $(PRECOMMIT_CONFIG_PATH)
```
Comment (li-boxuan) [inline Makefile:175]:
```suggestion
	@poetry run pre-commit run --files opendevin/**/* agenthub/**/* evaluation/**/* --show-diff-on-failure --config $(PRECOMMIT_CONFIG_PATH)
```

**Review by li-boxuan — APPROVED** [review]
Thanks, I completely forgot about that. Suggested adding `evaluation` folder. Feel free to not include it if there's too much diff.

**Comment by iFurySt** [comment]
> Thanks, I completely forgot about that. Suggested adding `evaluation` folder. Feel free to not include it if there's too much diff.

great. I've added the `evaluation`.

**Review by li-boxuan — APPROVED** [review]

File: evaluation/logic_reasoning/instruction.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
-You are a helpful assistant assigned with logic reasoning task. You need to determine the correctness of a query given some facts and fules. 
+You are a helpful assistant assigned with logic reasoning task. You need to determine the correctness of a query given some facts and fules.
```
Comment (tobitege) [inline evaluation/logic_reasoning/instruction.txt:1]: fules -> rules

**Review by tobitege — COMMENTED** [review]
could fix one typo

File: evaluation/logic_reasoning/instruction.txt (hunk 3 above)
Comment (li-boxuan) [inline evaluation/logic_reasoning/instruction.txt:1]: It's fine to fix this given that we haven't run benchmarks for logic reasoning systematically yet.

```suggestion
You are a helpful assistant assigned with logic reasoning task. You need to determine the correctness of a query given some facts and rules.
```

**Pull request merged**